src/main/java/com/autonomic/tmc/auth/exception/BaseSdkException.java

```diff
@@ -0,0 +1,78 @@
+/*-
+ * ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+ * TMC Auth SDK
+ * ——————————————————————————————————————————————————————————————————————————————
+ * Copyright (C) 2016 - 2020 Autonomic, LLC
+ * ——————————————————————————————————————————————————————————————————————————————
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License
+ * ______________________________________________________________________________
+ */
+package com.autonomic.tmc.auth.exception;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+
+@Slf4j
+public class BaseSdkException extends RuntimeException {
+
+    static Project project = new Project();
+
+    private final ErrorSourceType errorSourceType;
+
+    BaseSdkException(ErrorSourceType errorSourceType, String message) {
+        super(message);
+        this.errorSourceType = errorSourceType;
+    }
+
+    BaseSdkException(ErrorSourceType errorSourceType, String message, Throwable cause) {
+        super(message, cause);
+        this.errorSourceType = errorSourceType;
+    }
+
+    @Override
+    public String getMessage() {
+        return String.format("%s-%s-%s: %s", project.getName(), project.getVersion(), errorSourceType,
+                super.getMessage());
+    }
+
+    static class Project {
```

Comment:
Since this is distributed as a 'jar' should we be setting and getting this info from the `MANIFEST.MF` - I am caught up on the naming of "Project" as it is maven specifice vs "SDK_CLIENT or jar".

Reply:
How about the new name?